test/PunksVerifier.ts

```diff
@@ -0,0 +1,161 @@
+            user,
+            signers: signers.slice(2),
+        };
+    };
+
+    describe("verifyPredicates", () => {
+        let ctx: TestContext;
+
+        beforeEach(async () => {
+            ctx = await loadFixture(fixture);
+        });
+
+        it("fails for an invalid tokenId", async () => {
+            const { vaultFactory, user, verifier } = ctx;
+
+            const bundleId = await initializeBundle(vaultFactory, user);
+            const bundleAddress = await vaultFactory.instanceAt(bundleId);
+
+            // Will revert because 20000 is not a valid punk token Id
+            await expect(verifier.verifyPredicates(encodeInts([20000]), bundleAddress)).to.be.revertedWith("IV_InvalidTokenId");
+        });
+
+        it("verifies a specific punk token id", async () => {
+            const { vaultFactory, user, punks, verifier } = ctx;
+
+            // Start 2 bundles
+            const bundleId = await initializeBundle(vaultFactory, user);
+            const bundleAddress = await vaultFactory.instanceAt(bundleId);
+            const bundleId2 = await initializeBundle(vaultFactory, user);
+            const bundleAddress2 = await vaultFactory.instanceAt(bundleId2);
+
+            // Fund both bundles with different token IDs
+            const tokenId = 5555;
+            await punks.connect(user).getPunk(tokenId);
+            await punks.connect(user).transferPunk(bundleAddress, tokenId);
+
+            const tokenId2 = 7777;
+            await punks.connect(user).getPunk(tokenId2);
+            await punks.connect(user).transferPunk(bundleAddress2, tokenId2);
+
+            // First bundle should have item
+            expect(await verifier.verifyPredicates(encodeInts([tokenId]), bundleAddress)).to.be.true;
+            // Second bundle should not
+            expect(await verifier.verifyPredicates(encodeInts([tokenId]), bundleAddress2)).to.be.false;
+        });
+
+        it("verifies punks any token id", async () => {
+            const { vaultFactory, user, punks, verifier } = ctx;
+
+            // Start 3 bundles
+            const bundleId = await initializeBundle(vaultFactory, user);
+            const bundleAddress = await vaultFactory.instanceAt(bundleId);
+            const bundleId2 = await initializeBundle(vaultFactory, user);
+            const bundleAddress2 = await vaultFactory.instanceAt(bundleId2);
+            const bundleId3 = await initializeBundle(vaultFactory, user);
+            const bundleAddress3 = await vaultFactory.instanceAt(bundleId3);
+
+            // Fund both bundles with different token IDs
+            const tokenId = 5555;
+            await punks.connect(user).getPunk(tokenId);
+            await punks.connect(user).transferPunk(bundleAddress, tokenId);
+
+            const tokenId2 = 7777;
+            await punks.connect(user).getPunk(tokenId2);
+            await punks.connect(user).transferPunk(bundleAddress2, tokenId2);
+
+            // First and bundle should have item
```

Comment:
The word `second` is missing from comment.

Reply:
Fixed